### PR TITLE
refactor(infra): directory structure

### PR DESCRIPTION
**Pull request by Parkerhiphop**
## Why need this change? / Root cause:

- Change the directory structure by consensus.
- Current files are grouped by features(auth, lobby, ...), but group by type(requests, components, ... ) is more suitable for the projects.

## Changes made:

- Remove `shared/`, and move all the sub folders to root layer.
- Place atomic and highly reusable components into `components/shared/`
- Improve README.md
- Remove `.env` and add `.env.example`
- Prettier all impacted files

## Test Scope / Change impact:

-

## Issue

-

### CODE REVIEW

**Review by Yuwen-ctw — APPROVED** [review]

File: .env
Comment (Yuwen-ctw) [inline .env:1]: 想請問這裡移除 .env 後，目前 yarn dev 會得到 AxiosError: Request failed with status code 500 是預期的嗎?
是否因應開發環境，需要新建一個 .env.development 的檔案呢?

File: .env.development [hunk 1]
```diff
@@ -0,0 +1,2 @@
+INTERNAL_ENDPOINT=http://localhost:3030
+MOCK=true
```
Comment (Yuwen-ctw) [inline .env.development:2]: 此 commit 進行以下異動:
1. 新增開發環境變數檔。
2. 因應新的資料夾結構，調整 storybook 與 tailwind 的設定路徑。

**Review by JohnsonMao — APPROVED** [review]
LGTM

**Review by ttpss930141011 — APPROVED** [review]
LGTM

**Review by HongLinLiao — APPROVED** [review]

**Pull request merged**